spyder/app/tests/test_mainwindow.py

```diff
@@ -1437,6 +1437,28 @@ def test_run_cython_code(main_window, qtbot):
     main_window.editor.close_file()
 
 
+@pytest.mark.slow
+def test_project_path(main_window, tmpdir, qtbot):
+    """Test project path added to spyder_pythonpath."""
+    # Wait until the window is fully up
+    shell = main_window.ipyconsole.get_current_shellwidget()
+    qtbot.waitUntil(
+        lambda: shell._prompt_html is not None, timeout=SHELL_TIMEOUT)
+
+    projects = main_window.projects
+
+    # Create a project
+    path = str(tmpdir.mkdir('project_path'))
+
+    assert path not in projects.get_conf('spyder_pythonpath', section='main')
+
+    projects.open_project(path=path)
+
+    assert path in projects.get_conf('spyder_pythonpath', section='main')
+
```

Comment:
Please also test at this point that the project path was added to `sys.path` in the new IPython console that is created after the project was opened.

Reply:
I pushed up changes to test propagation to IPython consoles